toolchain/lower/function_context.cpp

```diff
@@ -98,6 +98,11 @@ auto FunctionContext::LowerInst(SemIR::InstId inst_id) -> void {
   CARBON_VLOG() << "Lowering " << inst_id << ": " << inst << "\n";
   builder_.getInserter().SetCurrentInstId(inst_id);
   auto loc = file_context_->GetDiagnosticLoc(inst_id);
+  CARBON_CHECK(loc.filename == di_subprogram_->getFile()->getFilename())
+      << "Instructions located in a different file from their enclosing "
+         "function aren't handled yet";
+  CARBON_CHECK(loc.line_number >= 1 && loc.column_number >= 1)
+      << "Instructions without line locations aren't handled yet";
```

Comment:
FYI, I think this isn't catching the existing issues because it's not used everywhere GetDiagnosticLoc is.

Reply:
To be sure, what I was noting here was that file_context.cpp also queries ConvertLoc. Since it won't use this code, the function missing a line number could be missed. That is, this would miss:

```
// CHECK:STDOUT: !4 = distinct !DISubprogram(name: "__global_init", scope: null, file: !3, line: 4294967295, type: !5, spFlags: DISPFlagDefinition, unit: !2)
```

#4252 fixes this.